### DISCUSSION TITLE
GitHub Secret in Renovate config JSON

### DISCUSSION THREAD

**Discussion by jamesdh** [renovatebot/renovate#14381] [Request Help]
Trying to setup maven dependency resolution against our own private GitHub Package Repository using our own self-hosted (via GitHub Actions) runner. This requires defining a [hostRule](https://docs.renovatebot.com/configuration-options/#hostrules) in the config JSON that includes a personal access token. Is there any way to pass in that token (via a GitHub Secret) so that it isn't stored in clear text?

**Comment by rarkins** [comment]
Look up the docs for "encrypted"

**Reply by jamesdh** [reply]
That gives me [this](https://docs.renovatebot.com/configuration-options/#encrypted) which isn't very informative I'm afraid. 

The only other thing I can find is [Adding an encrypted npm token to Renovate Config](https://docs.renovatebot.com/getting-started/private-packages/#add-an-encrypted-npm-token-to-renovate-config) but that appears to be dependent on using the hosted version?

**Comment by jamesdh** [comment]
Stumbled upon [detectHostRulesFromEnv](https://docs.renovatebot.com/self-hosted-configuration/#detecthostrulesfromenv) which sounds like it would allow for this to work
